api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -131,40 +219,16 @@ public UnboundPredicate<Integer> projectStrict(String name, BoundPredicate<Integ
       }
       return null;
     }
-
-    @Override
-    public boolean equals(Object o) {
-      if (this == o) {
-        return true;
-      } else if (!(o instanceof TruncateInteger)) {
-        return false;
-      }
-
-      TruncateInteger that = (TruncateInteger) o;
-      return width == that.width;
-    }
-
-    @Override
-    public int hashCode() {
-      return Objects.hashCode(width);
-    }
-
-    @Override
-    public String toString() {
-      return "truncate[" + width + "]";
-    }
   }
 
-  private static class TruncateLong extends Truncate<Long> {
-    private final int width;
-
+  private static class TruncateLong extends Truncate<Long> implements Function<Long, Long> {
     private TruncateLong(int width) {
-      this.width = width;
+      super(width);
     }
 
     @Override
-    public Integer width() {
-      return width;
+    public Function<Long, Long> bind(Type type) {
+      return this;
```

Comment:
Is it different compared to how we handle integers above?

Reply:
The only difference is that this function operations on longs rather than ints.